### PR TITLE
Combine Botocore Tests

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Combine boto3, botocore, and bedrock tests into 1 test suite for performance.
  * The logical separation of boto3 from botocore only makes sense in the event of having separate instrumentation. Since we lack that, botocore is really the only library with hooks under test. boto3 tests are simply integration tests with botocore.
  * Bedrock was never truly separate, just a different instrumentation effort than the external calls previously instrumented for botocore.

### CODE REVIEW

**Comment by TimPansino** [comment]
Open question: should any adjustments be made to the existing matrixing? Currently boto3 and bedrock tests were only targeting botocore-latest. Should we revisit the matrixing on these combined tests for botocore?

File: tests/external_botocore/conftest.py [hunk 1]
```diff
@@ -12,19 +12,142 @@
...
+    "debug.record_transaction_failure": True,
+    "ml_insights_events.enabled": True,
 }
-
 collector_agent_registration = collector_agent_registration_fixture(
-        app_name='Python Agent Test (external_botocore)',
-        default_settings=_default_settings)
+    app_name="Python Agent Test (external_botocore)",
+    default_settings=_default_settings,
+    linked_applications=["Python Agent Test (external_botocore)"],
```
Comment (hmstepanek) [inline tests/external_botocore/conftest.py:43]: Does this linked_applications need to be here?

**Review by hmstepanek — APPROVED** [review]
Looks good I just had one question about whether the linked_applications was necessary.

**Review by hmstepanek — APPROVED** [review]
Looks good now. I just merged the claude model in and I merged that change into this branch. Assuming the tests pass (ignoring starlette failing) let's go ahead and merge this.

**Comment by TimPansino** [comment]
> Looks good now. I just merged the claude model in and I merged that change into this branch. Assuming the tests pass (ignoring starlette failing) let's go ahead and merge this.

Only hypercorn and starlette are failing, going to merge this.

**Pull request merged**